test/core/miner/PreCommitBatcherTest.cpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include <miner/storage_fsm/types.hpp>
+#include "miner/storage_fsm/impl/precommit_batcher_impl.hpp"
+#include "testutil/literals.hpp"
+#include "testutil/mocks/libp2p/scheduler_mock.hpp"
+
+namespace fc::mining {
+  using api::MinerInfo;
+  using api::SignedMessage;
+  using api::UnsignedMessage;
+  using fc::mining::types::DealInfo;
+  using fc::mining::types::PaddedPieceSize;
+  using fc::mining::types::Piece;
+  using fc::mining::types::PieceInfo;
+  using libp2p::protocol::SchedulerMock;
+  using libp2p::protocol::scheduler::Ticks;
+  using primitives::tipset::Tipset;
+  using primitives::tipset::TipsetCPtr;
+  using testing::Mock;
+
+  class PreCommitBatcherTest : public testing::Test {
+   protected:
+    virtual void SetUp() {
+      seal_proof_type_ = RegisteredSealProof::kStackedDrg2KiBV1;
+      api_ = std::make_shared<FullNodeApi>();
+      sch_ = std::make_shared<SchedulerMock>();
+      miner_id = 42;
+      miner_address_ = Address::makeFromId(miner_id);
+      current_time_ = toTicks(std::chrono::duration_cast<std::chrono::seconds>(
+          std::chrono::system_clock::now().time_since_epoch()));
+
+      api::BlockHeader block;
+      block.height = 2;
+
+      auto tipset = std::make_shared<Tipset>(
+          TipsetKey(), std::vector<api::BlockHeader>({block}));
+
+      api_->ChainHead = [=]() -> outcome::result<TipsetCPtr> {
+        return outcome::success(tipset);
+      };
+
+      api_->StateMinerInfo =
+          [=](const Address &address,
+              const TipsetKey &tipset_key) -> outcome::result<MinerInfo> {
+        if (address == miner_address_) {
+          MinerInfo info;
+          info.seal_proof_type = seal_proof_type_;
+          return info;
+        }
+        return ERROR_TEXT("Error");
+      };
+
+      EXPECT_CALL(*sch_, now()).WillOnce(testing::Return(current_time_));
+      auto result = PreCommitBatcherImpl::makeBatcher(
+          toTicks(std::chrono::seconds(60)), api_, sch_, miner_address_);
+      batcher_ = result.value();
+      ASSERT_FALSE(result.has_error());
+    }
+
+    std::shared_ptr<FullNodeApi> api_;
+    std::shared_ptr<SchedulerMock> sch_;
+    std::shared_ptr<PreCommitBatcherImpl> batcher_;
+    Address miner_address_;
+    uint64_t miner_id;
+    RegisteredSealProof seal_proof_type_;
+    Ticks current_time_;
+    primitives::TokenAmount mutualDeposit = 0;
+  };
+
+  TEST_F(PreCommitBatcherTest, BatcherWrite) {
+    SectorInfo si = SectorInfo();
+    api::SectorPreCommitInfo precInf;
+    primitives::TokenAmount deposit = 10;
+    ASSERT_FALSE(batcher_->addPreCommit(si, deposit, precInf).has_error());
+  };
+
+  /**
+   * CallbackSend test is checking that after the scheduled time for a Precommit
+   * collecting, all the stored batcher's data will be published in a
+   * messagePool
+   */
```

Comment:
We use notation 
given - what we have 
when  - what will be done
then  - what will a result

Reply:
fixed